app/controllers/api/order_cycles_controller.rb

```diff
@@ -15,26 +17,36 @@ def products
 
       render json: products
     rescue ProductsRenderer::NoProducts
-      render status: :not_found, json: ''
+      render_no_products
     end
 
     def taxons
-      taxons = Spree::Taxon.
-        joins(:products).
-        where(spree_products: { id: distributed_products }).
-        select('DISTINCT spree_taxons.*')
+      taxons = Rails.cache.fetch("oc-taxons-#{order_cycle.id}", expires_in: 30.seconds) do
+        Spree::Taxon.
+          joins(:products).
+          where(spree_products: { id: distributed_products }).
+          select('DISTINCT spree_taxons.*')
+      end
```

Comment:
If we go ahead with this, all the caching logic needs to be extracted to services. All calls to Rails.cache need to be in a cache specific class, in this case for example: have a cached_data service that takes the cache key and yields back to this method and is re-used for the different places where we do this caching?